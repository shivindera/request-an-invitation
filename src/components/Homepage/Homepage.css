.broccoli-homepage {
  padding: 80px 16px 88px;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background: url('../../images/background.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: rgb(255 255 255 / 30%);
  background-blend-mode: lighten;
}

.broccoli-homepage-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: relative;
}

.broccoli-homepage-container h2 {
  font-size: 4rem;
  text-align: center;
}

.broccoli-homepage-container p {
  font-size: 2rem;
  padding: 1em 0;
  text-align: center;
}
