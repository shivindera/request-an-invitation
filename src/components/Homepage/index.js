// Homepage component, shown in between the Header and the Footer

import { useState } from 'react';
import RequestInvite from '../RequestInvite';
import RequestInviteSuccess from '../RequestInviteSuccess';
import { Button, Popup } from '../../utils';
import './Homepage.css';

const Homepage = () => {
  // Flag used to open close the Invite popup
  const [invitePopup, setInvitePopup] = useState(false);
  // Flag used to change the type of popup (Invite Form vs Invited Successfully)
  const [successPopup, setSuccessPopup] = useState(false);

  // Reset state of flags when popup is closed
  const handleDone = () => {
    setInvitePopup(false);
    setSuccessPopup(false);
  };

  return (
    <div className='broccoli-homepage'>
      <div className='broccoli-homepage-container'>
        <h2>
          A better way
          <br />
          to enjoy every day.
        </h2>
        <p>Be the first to know when we launch</p>
        <Button type='primary' onClick={() => setInvitePopup(true)}>
          Request an invite
        </Button>
      </div>
      <Popup isOpen={invitePopup} onClose={handleDone}>
        {successPopup ? (
          <RequestInviteSuccess onDone={handleDone}></RequestInviteSuccess>
        ) : (
          <RequestInvite onSubmitSuccess={() => setSuccessPopup(true)}></RequestInvite>
        )}
      </Popup>
    </div>
  );
};

export default Homepage;
