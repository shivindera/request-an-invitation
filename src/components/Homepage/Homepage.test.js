import { fireEvent, render, screen } from '@testing-library/react';
import Homepage from './';

it('renders Header as expected', () => {
  const homepage = render(<Homepage />);
  expect(homepage.container).toMatchSnapshot();
});

it('Request an invite buttom opens popup as expected', () => {
  const homepage = render(<Homepage />);
  fireEvent.click(homepage.getByText('Request an invite'));
  expect(screen.getByText('Send!')).toBeDisabled();
});
