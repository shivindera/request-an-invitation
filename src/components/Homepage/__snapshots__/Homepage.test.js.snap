// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header as expected 1`] = `
<div>
  <div
    class="broccoli-homepage"
  >
    <div
      class="broccoli-homepage-container"
    >
      <h2>
        A better way
        <br />
        to enjoy every day.
      </h2>
      <p>
        Be the first to know when we launch
      </p>
      <button
        class="broccoli-button broccoli-button--primary"
        type="button"
      >
        Request an invite
      </button>
    </div>
  </div>
</div>
`;
